cmake/modules/FindZephyr-sdk.cmake

```diff
@@ -123,16 +123,27 @@ if(("zephyr" STREQUAL ${ZEPHYR_TOOLCHAIN_VARIANT}) OR
         endif()
       endforeach()
     else()
+      if("${Zephyr-sdk_FIND_VERSION_RANGE_MAX}" STREQUAL "INCLUDE")
+        set(upper_bound _EQUAL)
+      endif()
+
+      if(NOT DEFINED Zephyr-sdk_FIND_VERSION_RANGE)
+        # Range not given, max out to ensure max version is not in effect.
+        set(Zephyr-sdk_FIND_VERSION_MAX 99999999)
+      endif()
+
       # Loop over each found Zepher SDK version until one is found that is compatible.
       foreach(zephyr_sdk_candidate ${Zephyr-sdk_CONSIDERED_VERSIONS})
-        if("${zephyr_sdk_candidate}" VERSION_GREATER_EQUAL "${Zephyr-sdk_FIND_VERSION}")
+        if("${zephyr_sdk_candidate}" VERSION_GREATER_EQUAL "${Zephyr-sdk_FIND_VERSION}"
+           AND "${zephyr_sdk_candidate}" VERSION_LESS${upper_bound} "${Zephyr-sdk_FIND_VERSION_MAX}"
```

Comment:
missing space?

Reply:
no, this is on purpose so that the check expands to either: `VERSION_LESS` or `VERSION_LESS_EQUAL` depending on whether `Zephyr-sdk_FIND_VERSION_RANGE_MAX` is `INCLUDE` (--> `VERSION_LESS_EQUAL`) or `EXCLUDE` (--> `VERSION_LESS`).

Reply:
wasn't even aware you could do that, neat feature